### PR TITLE
Optional services

### LINKED ISSUE #387: Make services more configurable at runtime and dump-time
## Is your feature request related to a problem? Please describe.

placeholder:
@gabe-l-hart has some deets

## Additional context

Add any other context about the feature request here.

### PR DESCRIPTION
**Pull request by gabe-l-hart**
**What this PR does / why we need it**:

Closes #387 

This PR makes it possible to enable/disable the auto-generated inference and training services in both the http and grpc servers

### CODE REVIEW

**Review by prashantgupta24 — APPROVED** [review]
LGTM!

**Comment by gabe-l-hart** [comment]
Hmm, something about this PR seems to reliably trigger the 3.11 tests to deadlock on `test_threads_canceled_when_interrupt_fails`. I'll dig a little further.

**Comment by prashantgupta24** [comment]
@gabe-l-hart I see 3.11 failures on other PRs too

**Comment by gabe-l-hart** [comment]
Yeah, there's something screwy going on, but they seem a lot more consistent with this PR. Still digging

File: caikit/runtime/servicers/global_predict_servicer.py [hunk 1]
```diff
@@ -97,6 +97,8 @@ def __init__(
     ):
         self._started_metering = False
         self._model_manager = ModelManager.get_instance()
+        self._started_metering = False
```
Comment (prashantgupta24) [inline caikit/runtime/servicers/global_predict_servicer.py:100]: duplicate `_started_metering`

Comment (prashantgupta24) [inline caikit/runtime/servicers/global_predict_servicer.py:100]:
```suggestion
```

**Pull request merged**